### PR TITLE
perf(rspack_plugin_javascript): use Rayon to parse modules parallelly

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Improve perf issue introduced in #6697. Use Rayon to parallelly parse modules.

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by fi3ework** [comment]
!bench

**Comment by fi3ework** [comment]
The performance regression relieved than before.

**Comment by fi3ework** [comment]
!bench

File: crates/rspack_plugin_javascript/src/plugin/mod.rs [hunk 1]
```diff
@@ -786,150 +795,203 @@ impl JsPlugin {
   ) -> Result<HashMap<String, Arc<dyn Source>>> {
     let mut inlined_modules_to_info: HashMap<String, InlinedModuleInfo> = HashMap::new();
```
Comment (h-a-n-a) [inline crates/rspack_plugin_javascript/src/plugin/mod.rs:796]: I was wondering if the key can be a direct `Identifier`. Its performance is supposed to be better than `string`.

Comment (fi3ework) [reply crates/rspack_plugin_javascript/src/plugin/mod.rs]: Done. Good suggestion.

**Comment by fi3ework** [comment]
!bench

**Comment by fi3ework** [comment]
!bench

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**